### ISSUE TITLE
Fatal Error with Yoast SEO version 8

### ISSUE THREAD

**Issue by Mahjouba91** [timber/timber#1775]
### Expected behavior
No Fatal error when using Timber and Yoast SEO

### Actual behavior
Catchable fatal error : 
Argument 1 passed to WPSEO_Admin_Bar_Menu::get_post_focus_keyword() must be an instance of WP_Post, instance of Timber\Post given, called in plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php on line 204 and defined in plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php on line 474.

### Steps to reproduce behavior
Just visit a front page with admin bar

### What version of WordPress, PHP and Timber are you using?
WordPress : 4.9.6
PHP 5.6
Timber : 1.7.1

### How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)
Composer in the mu-plugins folder (with Bedrock stack)

**Comment by vhaas** [comment]
It seems like the type check is going to be fixed by Yoast SEO.
https://github.com/Yoast/wordpress-seo/pull/10679/files

**Comment by Mahjouba91** [comment]
Hi,

It seems that there is again a Fatal Error with Yoast version 8.3 and Timber : 

Fatal error: Uncaught TypeError: Argument 1 passed to WPSEO_Admin_Bar_Menu::get_post_focus_keyword() must be an instance of WP_Post, instance of Timber\Post given, called in web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php on line 204 and defined in web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php:474 Stack trace: #0 web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php(204): WPSEO_Admin_Bar_Menu->get_post_focus_keyword(Object(Timber\Post)) #1 web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php(61): WPSEO_Admin_Bar_Menu->add_keyword_research_submenu(Object(WP_Admin_Bar)) #2 web/wp/wp-includes/class-wp-hook.php(286): WPSEO_Admin_Bar_Menu->add_menu(Object(WP_Admin_Bar)) #3 web/wp/wp-includes/class-wp-hook.php(310): WP_Hook->apply_filters(NULL, Array) #4 web/wp/wp-includes/plugin.php(515): WP_Hook->do_act in web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php on line 474

**Comment by Mahjouba91** [comment]
@jarednova I don't know how to reopen the issue

**Comment by jarednova** [comment]
Re-opened! That said, this still seems like an issue that Yoast might have to resolve so they're a bit more permissive on the checking for a `WP_Post`. Any idea what's going on at `web/app/plugins/wordpress-seo/inc/class-wpseo-admin-bar-menu.php on line 204` that might have to change?

**Comment by palmiak** [comment]
@Mahjouba91 could you check older versions 8.2.1 or 8.1 ?

They also should fail. But let's be sure.

Also could you show us your code (the place where it fails).

Are you using any other plugins?

On the other hand I think that the problem is related to:
```php
protected function get_singular_post() {
	global $pagenow, $post;
	if ( ! is_singular() && ( ! is_blog_admin() || ! WPSEO_Metabox::is_post_edit( $pagenow ) ) ) {
		return null;
	}
	if ( ! isset( $post ) || ! is_object( $post ) ) {
		return null;
	}
	return $post;
}```

It may be Yoast fault (I'll try to reproduce it today - should be easy) or the problem may be in your code or in plugins you use. Some of it does something or create a post variable that isn't a WP_Post object.

**Comment by palmiak** [comment]
I tried to replicate your problem.

I used the same WP, Timber and Yoast SEO version. The only difference was the PHP. So either there is a problem with PHP (if you read the requirements of Yoast SEO plugin you can see `PHP version 7.2 or greater` - https://kb.yoast.com/kb/plugin-requirements/ - it may or may not work on older versions).

So I think we can narrow it to:
- PHP version
- plugins you use
- your code

I don't think it's on Timber side. Still - show us your code and plugins you use - let's find what's causes the problem.

BTW. if you can - try to update your PHP to 7.

**Comment by pascalknecht** [comment]
@Mahjouba91 @palmiak I checked this today as I also had this issue. And indeed it is Timbers fault in some kind. As the global `$post` variable is overwritten when you take the starter theme this occurs as Yoast did a typecheck of the `$post` variable. They have removed it now in master and this will probably be in the next release. https://github.com/Yoast/wordpress-seo/commit/280035d7d62dc5818fa425db00df9c150cfbbe79#diff-bf82b1227eaf9b2a3fe46f75bec88476

Closing this issue as no further actions are neede.
